root_profiler/engine/db.php

```diff
@@ -133,50 +106,44 @@ function query($query = "", $transaction = FALSE)
 		{
 			if( $this->in_transaction )
 			{
-				mysql_query("ROLLBACK", $this->db_connect_id);
-				$this->in_transaction = FALSE;
+				$this->dbh->exec("ROLLBACK");
+				$this->in_transaction = false;
 			}
 			return false;
 		}
 	}
 
 	function num_rows()
 	{
-		return ( $this->db_connect_id ) ? mysql_affected_rows($this->db_connect_id) : false;
+		return $this->query_result->rowCount();
 	}
 
-        function fetch_row($query_id = 0)
-        {
-                if( !$query_id )
-                {
-                        $query_id = $this->query_result;
-                }
-
-                if( $query_id )
-                {
-                        $this->row[$query_id] = mysql_fetch_array($query_id, MYSQL_ASSOC);
-                        return $this->row[$query_id];
-                }
-                else
-                {
-                        return false;
-                }
-        }
-
-	function freeresult($query_id = 0)
+	function fetch_row($stmt = null)
 	{
-		if( !$query_id )
+		if( !$stmt )
 		{
-			$query_id = $this->query_result;
+			$stmt = $this->query_result;
 		}
 
-		if ( $query_id )
+		if( $stmt )
 		{
-			unset($this->row[$query_id]);
-			unset($this->rowset[$query_id]);
+			return $stmt->fetch(PDO::FETCH_ASSOC);
+		}
+		else
```

Comment:
No need for an else clause when the if's block ends in a return.  (Could be justified if the else were "symmetric" to the if in some way, but this is clearly just the fall-back.)